### PR TITLE
create local directory hierarchy

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

**Comment by robert-chiniquy** [comment]
+1

File: file_syncer/syncer.py [hunk 1]
```diff
@@ -345,6 +345,11 @@ def _download_remote_file(self, name):
 
         driver = self._get_driver_instance()
         filepath = os.path.join(self._directory, local_filename)
+        dirname = os.path.dirname(filepath)
+
+        # make sure the path exists
```
Comment (Kami) [inline file_syncer/syncer.py:350]: `dirname` will be an empty string if you call `os.path.dirname` on a file and not a directory.

To correctly handle this case I would change the if to:

``` python
if dirname and not os.path.exists(dirname)
```

Comment (rphillips) [inline file_syncer/syncer.py:350]: Thanks for the tip... fixed.

**Comment by Kami** [comment]
Published a new version (0.3.2) on PyPi - https://pypi.python.org/pypi/file_syncer/0.3.2

Thanks!

**Pull request merged**